### PR TITLE
Self.__functions

### PR DESCRIPTION
**Pull request by wiljnichepa**
Hey @dpgraham4401, this is an extension of the design you implemented for the GET functions to POST, PUT, and DELETE. There are a few outstanding functions that I've not adapted to this structure, but those have different requirements - mainly passing or receiving attachments. I haven't figured out if it's worth it to write a new master function for those when there may only be 1-2 functions relying on it, thus overcomplicating it unnecessarily.

I've also made a few formatting fixes and updated the variable names when making requests() calls to be **resp** consistently.

### CODE REVIEW

File: emanifest-py/src/emanifest/client.py [hunk 1]
```diff
@@ -906,3 +832,32 @@ def __RCRAGet(self, endpoint):
...
+                            headers={'Accept': 'application/json','Authorization': 'Bearer ' + self.token})
+        if resp.ok:
+            return resp.json()
+        else:
+            print('Error: ' + str(resp.json()['message']))
+            
+            
+    def __RCRAPut(self, endpoint):
+        resp = requests.put(endpoint,
+                            headers={'Content-Type': m.content_type, 'Accept': 'application/json',
```
Comment (dpgraham4401) [inline emanifest-py/src/emanifest/client.py:857]: I think the object "m" here is an undefined reference

File: emanifest-py/src/emanifest/client.py [hunk 2]
```diff
@@ -767,10 +707,10 @@ def GetAttachmentsReg(self, mtn):
             attachments: PDF and HTML files containing additional manifest information (such as scans or electronic copies) for the given MTN
               message of success or failure
         """
-        attach = requests.get(self.base_url + '/api/v1/state/emanifest/manifest/' + mtn + '/attachments',
+        resp = requests.get(self.base_url + '/api/v1/state/emanifest/manifest/' + mtn + '/attachments',
                               headers={'Accept': 'multipart/mixed', 'Authorization': 'Bearer ' + self.token},
                               stream=True)
-        if attach.ok:
+        if resp.ok:
             multipart_data = decoder.MultipartDecoder.from_response(attach)
```
Comment (dpgraham4401) [inline emanifest-py/src/emanifest/client.py:714]: "attach needs to be changed to "resp" I believe
```suggestion
            multipart_data = decoder.MultipartDecoder.from_response(resp)
```

File: emanifest-py/src/emanifest/client.py [hunk 3]
```diff
@@ -783,7 +723,6 @@ def GetAttachmentsReg(self, mtn):
         else:
             print('Error: ' + str(attach.json()['message']))
```
Comment (dpgraham4401) [inline emanifest-py/src/emanifest/client.py:724]:
```suggestion
            print('Error: ' + str(resp.json()['message']))
```

File: emanifest-py/src/emanifest/client.py (hunk 1 above)
Comment (wiljnichepa) [reply emanifest-py/src/emanifest/client.py:857]: Good catch - forgot to add that to the parameters. I've updated the function to require it.

Comment (dpgraham4401) [inline emanifest-py/src/emanifest/client.py:857]: I think the object "m" here is an undefined reference, I'm not sure what it's supposed to be.

**Review by dpgraham4401 — CHANGES_REQUESTED** [review]
There's a couple undefined references we should correct before merging

File: emanifest-py/src/emanifest/client.py (hunk 1 above)
Comment (wiljnichepa) [reply emanifest-py/src/emanifest/client.py:857]: It's the multipartencoder value necessary to pass the zip/json files. We need both the underlying data and the content-type to successfully PUT it. Like *endpoint*, it is defined in the correct/update functions. I've modified the function in another commit to require m as a parameter.

**Pull request merged**